### PR TITLE
Temp bu

### PR DESCRIPTION
**Pull request by saiprasad1997**

### CODE REVIEW

File: src/app/app-routing.module.ts [hunk 1]
```diff
@@ -62,19 +59,22 @@ export const routes: Routes = [
...
-    component : UserMappingComponent,
+    path : 'create-subscription',
+    component : CreateSubscriptionComponent,
     canActivate: [AuthGuard]
   },
   {
     path : 'saved-mapping',
-    component : SavedMappingComponent,
-    canActivate: [AuthGuard]
+    component : SavedMappingComponent
```
Comment (Susmith-roy-incontact) [inline src/app/app-routing.module.ts]: use authguard

File: src/app/app-routing.module.ts [hunk 2]
```diff
@@ -93,7 +93,16 @@ export const routes: Routes = [
   {
     path : 'tenants/tenant-details',
     component : BuDetailsWithUserMappingComponent,
```
Comment (Susmith-roy-incontact) [inline src/app/app-routing.module.ts]: use authguard atleast for parent and use for child too if possible

File: src/app/bu/create-ic-bu/component/create-ic-bu.component.html [hunk 3]
```diff
@@ -1,9 +1,41 @@
...
+        </div>
+      </div>
+    </div>
+  </div>
+</div>
+
 <div *ngIf="showContentIncontactForm">
 <div class="create-ic-bu-data">
-  <form [formGroup]="form" (ngSubmit)="onSubmit()">
+  <form [formGroup]="form2" (ngSubmit)="onSubmit()">
```
Comment (pooja-koul-incontact) [inline src/app/bu/create-ic-bu/component/create-ic-bu.component.html]: give more informative name

File: src/app/bu/create-ic-bu/component/create-ic-bu.component.html [hunk 4]
```diff
@@ -1,9 +1,41 @@
...
   <app-loading-spinner></app-loading-spinner>
 </div>
+
+<div class="create-incontact-bu-error-modal" *ngIf="getModalDialog()">
+  <div class="modal-fade" tabindex="-1" role="dialog">
+    <div class="breeze b-error modal-dialog" role="document">
+      <div class="modal-content">
+        <div class="modal-header">
+          <button type="button" class="close" data-dismiss="modal" aria-label="Close" (click)="onClickDismissModal()">
+            <svg fill="#afc1cf" class="nb-icon-close" viewBox="0 0 20 20" width="22" height="22">
```
Comment (Susmith-roy-incontact) [inline src/app/bu/create-ic-bu/component/create-ic-bu.component.html]: can we keep the svgs in assets folder and refer them when needed , if in future someone need to modify svg ,he need to find all the places where it is used and modify accordingly

File: src/app/bu/create-ic-bu/component/create-ic-bu.component.ts [hunk 5]
```diff
@@ -4,63 +4,79 @@ import {Router} from '@angular/router';
...
 
 @Component({
   selector: 'app-create-ic-bu',
   templateUrl: './create-ic-bu.component.html',
   styleUrls: ['./create-ic-bu.component.css']
 })
 export class CreateIcBuComponent implements OnInit {
-  buObject = new CreateBUData();
-  form: FormGroup;
+  buObject2 = { buName: '',
```
Comment (pooja-koul-incontact) [inline src/app/bu/create-ic-bu/component/create-ic-bu.component.ts]: again, the name should be more informative

File: src/app/bu/create-partner-bu/component/create-partner-bu.component.html [hunk 6]
```diff
@@ -1,6 +1,39 @@
...
   <app-loading-spinner></app-loading-spinner>
 </div>
+
+<div class="create-partner-bu-error-modal" *ngIf="getModalDialog()">
+<div class="modal-fade" tabindex="-1" role="dialog">
+  <div class="breeze b-error modal-dialog" role="document">
+    <div class="modal-content">
+      <div class="modal-header">
+        <button type="button" class="close" data-dismiss="modal" aria-label="Close" (click)="onClickDismissModal()">
+          <svg fill="#afc1cf" class="nb-icon-close" viewBox="0 0 20 20" width="22" height="22">
```
Comment (Susmith-roy-incontact) [inline src/app/bu/create-partner-bu/component/create-partner-bu.component.html]: keep the svg in assets folder

File: src/app/bu/display-mapped-bu/component/display-mapped-bu.component.html [hunk 7]
```diff
@@ -1,3 +1,34 @@
...
+      <div class="modal-content">
+        <div class="modal-header">
+          <button type="button" class="close" data-dismiss="modal" aria-label="Close" (click)="onClickDismissModal()">
+            <svg fill="#afc1cf" class="nb-icon-close" viewBox="0 0 20 20" width="22" height="22">
+              <title>Close</title>
+              <path d="M11.4 10l4.7-4.7c0.4-0.4 0.4-1 0-1.4s-1-0.4-1.4 0l-4.7 4.7-4.7-4.8c-0.4-0.4-1-0.4-1.4 0s-0.4 1 0 1.4l4.7 4.8-4.7 4.7c-0.4 0.4-0.4 1 0 1.4s1 0.4 1.4 0l4.7-4.7 4.7 4.7c0.4 0.4 1 0.4 1.4 0s0.4-1 0-1.4l-4.7-4.7z"></path>
+            </svg>
+          </button>
+          <h4 class="modal-title module-title" id="myModalLabel">
+            <svg fill="#fa3f40" class="nb-modal-icon" viewBox="0 0 20 20" width="25" height="25">
```
Comment (Susmith-roy-incontact) [inline src/app/bu/display-mapped-bu/component/display-mapped-bu.component.html]: keep svg in assets

File: src/app/bu/map-bu/component/map-bu.component.ts [hunk 8]
```diff
@@ -30,11 +35,33 @@ export class MapBuComponent implements OnInit {
...
-
-    });
 
-    this.router.navigate(['tenants']);
+    this.createBuService.createPartnerBU(this.createBuService.partnerBuRequestObj).subscribe(
+      (response1: CreatePartnerTenantResponse) => {
+        this.createBuService.createICBU(this.createBuService.incontactBuRequestObj).subscribe(
+          (response2: CreateAcdTenantResponse) => {
+            this.createBuService.mapBU(this.mapBuObj).subscribe((res: MapBuResponse) => {
+              localStorage.setItem('buMappingId', res.uuid);
```
Comment (Susmith-roy-incontact) [inline src/app/bu/map-bu/component/map-bu.component.ts]: use session storage

File: src/app/bu/map-bu/component/map-bu.component.ts [hunk 9]
```diff
@@ -30,11 +35,33 @@ export class MapBuComponent implements OnInit {
   mapBu() {
     this.mapBuObj.partnerBUId = this.createBuService.objMapBu.partnerBUId;
     this.mapBuObj.icBUId = this.createBuService.objMapBu.icBUId;
-    this.createBuService.mapBU(this.mapBuObj).subscribe((res: Response) => {
-
-    });
 
-    this.router.navigate(['tenants']);
+    this.createBuService.createPartnerBU(this.createBuService.partnerBuRequestObj).subscribe(
+      (response1: CreatePartnerTenantResponse) => {
```
Comment (Susmith-roy-incontact) [inline src/app/bu/map-bu/component/map-bu.component.ts]: give some meaningful name

File: src/app/bu/map-bu/component/map-bu.component.ts [hunk 10]
```diff
@@ -30,11 +35,33 @@ export class MapBuComponent implements OnInit {
...
+        this.createBuService.createICBU(this.createBuService.incontactBuRequestObj).subscribe(
+          (response2: CreateAcdTenantResponse) => {
+            this.createBuService.mapBU(this.mapBuObj).subscribe((res: MapBuResponse) => {
+              localStorage.setItem('buMappingId', res.uuid);
+              this.createBuService.setUUIDToBeSentToUserMapping(res.uuid);
+              this.router.navigate(['saved-mapping']);
+            },
+              error3 => {
```
Comment (Susmith-roy-incontact) [inline src/app/bu/map-bu/component/map-bu.component.ts]: some meaningful name to error3,error 2 error1

File: src/app/subscription/create-subscription/create-subscription.component.html [hunk 11]
```diff
@@ -1,3 +1,34 @@
...
+      <div class="modal-content">
+        <div class="modal-header">
+          <button type="button" class="close" data-dismiss="modal" aria-label="Close" (click)="onClickDismissModal()">
+            <svg fill="#afc1cf" class="nb-icon-close" viewBox="0 0 20 20" width="22" height="22">
+              <title>Close</title>
+              <path d="M11.4 10l4.7-4.7c0.4-0.4 0.4-1 0-1.4s-1-0.4-1.4 0l-4.7 4.7-4.7-4.8c-0.4-0.4-1-0.4-1.4 0s-0.4 1 0 1.4l4.7 4.8-4.7 4.7c-0.4 0.4-0.4 1 0 1.4s1 0.4 1.4 0l4.7-4.7 4.7 4.7c0.4 0.4 1 0.4 1.4 0s0.4-1 0-1.4l-4.7-4.7z"></path>
+            </svg>
+          </button>
+          <h4 class="modal-title module-title" id="myModalLabel">
+            <svg fill="#fa3f40" class="nb-modal-icon" viewBox="0 0 20 20" width="25" height="25">
```
Comment (Susmith-roy-incontact) [inline src/app/subscription/create-subscription/create-subscription.component.html]: keep svg in assets

File: src/app/subscription/create-subscription/create-subscription.component.html [hunk 12]
```diff
@@ -1,3 +1,34 @@
...
+              <title>Error_message</title>
+              <path d="M10 0c-5.5 0-10 4.5-10 10s4.5 10 10 10 10-4.5 10-10-4.5-10-10-10zM10 18c-4.4 0-8-3.6-8-8s3.6-8 8-8 8 3.6 8 8-3.6 8-8 8z"></path>
+              <path d="M10 11v0c-0.6 0-1-0.4-1-1v-4c0-0.5 0.4-1 1-1v0c0.5 0 1 0.4 1 1v4c0 0.6-0.4 1-1 1z"></path>
+              <path d="M10 16v0c-0.6 0-1-0.4-1-1v-1c0-0.5 0.4-1 1-1v0c0.5 0 1 0.4 1 1v1c0 0.6-0.4 1-1 1z"></path>
+            </svg>
+            &nbsp;Error Message</h4>
+        </div>
+        <div class="modal-body">
+          Something went wrong.
```
Comment (pooja-koul-incontact) [inline src/app/subscription/create-subscription/create-subscription.component.html]: give a more descriptive message

File: src/app/subscription/create-subscription/create-subscription.component.spec.ts [hunk 13]
```diff
@@ -0,0 +1,30 @@
...
+    .compileComponents();
+  }));
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(CreateSubscriptionComponent);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+  });
+
+  it('should create', () => {
```
Comment (pooja-koul-incontact) [inline src/app/subscription/create-subscription/create-subscription.component.spec.ts]: add more tests to test html

**Pull request merged**